code/modules/mob/living/carbon/human/_species.dm

```diff
@@ -285,25 +285,28 @@ GLOBAL_LIST_EMPTY(features_by_species)
  * * organ_holder - carbon, the owner of the species datum AKA whoever we're regenerating organs in
  * * old_species - datum, used when regenerate organs is called in a switching species to remove old mutant organs.
  * * replace_current - boolean, forces all old organs to get deleted whether or not they pass the species' ability to keep that organ
+ * * replace_hazardous - boolean, forces all hazardous organs to deleted, if it is not the default for the species
  * * excluded_zones - list, add zone defines to block organs inside of the zones from getting handled. see headless mutation for an example
  * * visual_only - boolean, only load organs that change how the species looks. Do not use for normal gameplay stuff
  */
-/datum/species/proc/regenerate_organs(mob/living/carbon/organ_holder, datum/species/old_species, replace_current = TRUE, list/excluded_zones, visual_only = FALSE)
+/datum/species/proc/regenerate_organs(mob/living/carbon/organ_holder, datum/species/old_species, replace_current = TRUE, replace_hazardous = TRUE, list/excluded_zones, visual_only = FALSE)
```

Comment:
I have a feeling this shouldn't have `replace_hazardous` - it just muddies the purpose of the proc

Leave `regenerate_organs` to JUST handling species organs and such and have `/carbon/proc/regenerate_organs` handle removing any dangerous organs 

Like just via
```dm
for(var/obj/item/organ/organ as anything in organs)
	if(remove_hazardous && (organ.organ_flags & ORGAN_HAZARDOUS))
		qdel(organ )
```

Reply:
Actually why even include this in `regenerate_organs` at all? Just have `fully_heal` do this